### PR TITLE
Update types/node/v11 & v12

### PR DESCRIPTION
**Pull request by lekoaf**
* Fixes documentation for `concat`
* Adds `Buffer | Uint8Array` union to `concat`, `compare`, `equals` and `copy`

Solves #36253.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/buffer.html#buffer_class_method_buffer_concat_list_totallength
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by lekoaf** [comment]
FYI, I couldn't run `npm test` because "logrocket" wasn't in the "whitelist".

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Please update the node v12 typings as well, please keep in mind that v11 is not LTS and is unsupported at this point.

**Comment by lekoaf** [comment]
Sure, I’ll take look at that tomorrow.

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -241,9 +241,10 @@ interface Buffer extends Uint8Array {
     write(string: string, offset: number, length: number, encoding?: BufferEncoding): number;
     toString(encoding?: string, start?: number, end?: number): string;
     toJSON(): { type: 'Buffer', data: number[] };
-    equals(otherBuffer: Uint8Array): boolean;
-    compare(otherBuffer: Uint8Array, targetStart?: number, targetEnd?: number, sourceStart?: number, sourceEnd?: number): number;
-    copy(targetBuffer: Uint8Array, targetStart?: number, sourceStart?: number, sourceEnd?: number): number;
+    equals(otherBuffer: Buffer | Uint8Array): boolean;
```
Comment (Flarna) [inline types/node/globals.d.ts]: `Buffer | Uint8Array` should be not needed as `Buffer extends Uint8Array`.

File: types/node/globals.d.ts [hunk 2]
```diff
@@ -241,9 +241,10 @@ interface Buffer extends Uint8Array {
...
+    compare(otherBuffer: Buffer | Uint8Array, targetStart?: number, targetEnd?: number, sourceStart?: number, sourceEnd?: number): number;
+    concat(list: Buffer[] | Uint8Array[], totalLength?: number): Buffer;
```
Comment (Flarna) [inline types/node/globals.d.ts]: `concat` is a class method and not a instance method therefore its part of the constructor and should be removed here.

**Review by Flarna — CHANGES_REQUESTED** [review]
see other comments

**Pull request closed without merging**